Add in PackageKit

matthewbauer opened the pull request:
###### Motivation for this change

see NixOS/nix#233
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

groxxda commented on pkgs/development/libraries/appstream-glib/default.nix:
```diff
@@ -1,19 +1,23 @@
 
 stdenv.mkDerivation rec {
-  name = "appstream-glib-0.3.6";
+  name = "appstream-glib-0.5.11";
```
Why not update to 0.5.15? :)

matthewbauer replied:
I already had forgotten why I did that but it turns out that gnome-software 3.18 needs that version of appstream-glib.

groxxda commented on pkgs/development/libraries/appstream-glib/default.nix:
```diff
@@ -1,19 +1,23 @@
+    owner = "hughsie";
+    repo = "appstream-glib";
+    sha256 = "1rvfncm9z29h70pd718j73cd263g6yyxkxrg7zfzy0gj6wwzvhkh";
+    rev = "appstream_glib_0_5_11";
```
Nitpick: maybe use `lib.replaceString ["." "-"] ["_" "_"] name`

groxxda commented on nixos/modules/services/misc/packagekit.nix:
```diff
@@ -0,0 +1,63 @@
+
+      enable = mkOption {
+        type = types.bool;
+        default = true;
```
We should probably default to `false` - or just use `mkEnableOption "PackageKit"`

groxxda commented on pkgs/tools/package-management/packagekit/default.nix:
```diff
@@ -1,57 +1,53 @@
-{ stdenv, fetchurl, intltool, glib, pkgconfig, polkit, python, sqlite }:
+{ stdenv, fetchgit, intltool, glib, pkgconfig, polkit, python, sqlite, systemd
+, gobjectIntrospection, vala, gtk_doc, autoreconfHook, autoconf-archive
+, nixUnstable, boost
```
I think it's common practice to import nix and override from all-packages.nix: `callPackage ... { nix = nixUnstable; };`

nckx commented on nixos/modules/services/misc/packagekit.nix:
```diff
@@ -0,0 +1,63 @@
+    services.dbus.packages = [ pkgs.packagekit ];
+
+    systemd.services.packagekit = {
+      description = "PackageKit Daemon";
```
Bah. I know you're just following upstream here, but seeing a literal synonym of the filename (`packagekit.service`) pretending to be a ‘description’ makes me disproportionately grumpy.

I should get out more.

matthewbauer replied:
Maybe we can just use the provided systemd file? I can't remember why I didn't use it originally though.

nckx replied:
The provided file contains the same Description, so my nitpicking certainly wasn't aimed at you. Just something to do while waiting for this to build so I could try it out.

(That failed — `fatal: reference is not a tree` — but I see there's a new one now. Re-compiling...)

matthewbauer commented:
This should be ready to merge now.

matthewbauer commented:
I'll rebase onto gnome 3.20 later today.

matthewbauer committed:
appstream-glib: 0.3.6 -> 0.5.11

- needed for gnome-software to build

matthewbauer committed:
packagekit: add latest from hughsie's github repo

- currently pulled in from Git until the next release of PackageKit
  has Nix support
- also: add in a service module to start packagekit properly
- nixos service can be enabled via services.packagekit.enable
- packagekit requires nixunstable to build properly

matthewbauer committed:
gnome3.18: use packagekit bindings for gnome-software

- init gnome-software for gnome3 at 3.18.3
- list gnome-software as an "optional package" for gnome3
- enable packagekit service when gnome3 is enabled

matthewbauer committed:
gnome-packagekit: init at 3.18.0

- provides useful tools for searching package
- included in gnome3 as an optional package

matthewbauer committed:
packagekit: add myself as maintainer

matthewbauer committed:
gnome3.20: add gnome-packagekit, gnome-software

Previously the two were just in gnome-3.18. This adds them as
optionalPackages in gnome3.20.

matthewbauer commented:
cc: @lethalman, @DamienCassou

DamienCassou commented on nixos/modules/services/misc/packagekit.nix:
```diff
@@ -0,0 +1,63 @@
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable PackageKit.";
```
What about being more verbose here to explain what is PackageKit and what this module does?

DamienCassou commented on nixos/modules/services/misc/packagekit.nix:
```diff
@@ -0,0 +1,63 @@
+      serviceConfig.Type = "dbus";
+    };
+
+    environment.etc."PackageKit/PackageKit.conf".text = packagekitConf;
```
Is it possible to give the configuration file path as argument to the command to avoid polluting `/etc`?

matthewbauer replied:
I can't find anything at runtime. I could set "sysconfdir" to a store file if you'd like. The annoying part is that it's also used by "gnome-software": https://github.com/GNOME/gnome-software/blob/98ab62364f93fd5dff15c54b9361198067836e8f/src/gs-vendor.c#L110.

DamienCassou commented on pkgs/desktops/gnome-3/3.18/misc/gnome-packagekit/default.nix:
```diff
@@ -0,0 +1,25 @@
+  buildInputs = [ libxslt gnome3.gtk packagekit fontconfig
+                  libcanberra_gtk3 libnotify dbus_glib dbus_libs ];
+
+  meta = with stdenv.lib; {
```
Could you please add a description here? Same for the other derivations.

matthewbauer commented:
Travis errors are saying "no space left on device". But, it's building find on my Linux box.

matthewbauer commented:
@nckx Could we get this merged fairly soon?

joachifm commented:
I don't see any objections to this and it looks superficially okay to me, so merging as is.
